src/mbgl/renderer/image_manager.cpp

```diff
@@ -125,12 +137,28 @@ void ImageManager::notifyIfMissingImageAdded() {
     }
 }
 
+void ImageManager::reduceMemoryUse() {
+    std::vector<std::string> unusedImages;
+    unusedImages.reserve(requestedImages.size());
+    for (auto& requestedImage : requestedImages)  {
+        if (requestedImage.second.empty() &&
+            images.find(requestedImage.first) != images.end()) {
+            unusedImages.push_back(requestedImage.first);
```

Comment:
why not call  removeImage(unusedImage); here and avoid allocation of unusedImages.reserve(requestedImages.size()); (L142)

Reply:
`removeImage` was modifying container that was iterated in for range loop, I refactored code to remove superfluous vector.